src/pyFAI/io/diffmap_config.py

```diff
@@ -0,0 +1,347 @@
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+"""Module with the configuration dataclass for diffraction mapping experiments"""
+
+__author__ = "Jérôme Kieffer"
+__contact__ = "Jerome.Kieffer@ESRF.eu"
+__license__ = "MIT"
+__copyright__ = "European Synchrotron Radiation Facility, Grenoble, France"
+__date__ = "04/04/2025"
+__status__ = "development"
+__docformat__ = 'restructuredtext'
+
+import os
+import time
+import copy
+import json
+import logging
+logger = logging.getLogger(__name__)
+from collections import namedtuple
+from .tree import TreeItem
+from .integration_config import dataclass, ClassVar, WorkerConfig, fields, asdict
+from .nexus import is_hdf5
+
+#constants:
+CURRENT_VERSION = 1  # former version were unassigned
+
+
+@dataclass
+class MotorRange:
+    """ This object represents a motor range
+
+    :param start: Begining of the movement
+    :param stop: End of the movement, included
+    :param step: Number of points (i.e. numberof steps + 1)
+    :param name: Name of the motor
+    """
+    start: float = 0.0
+    stop: float = 1.0
+    points: int = 0
+    name: str = ""
+
+    @classmethod
+    def _fromdict(cls, dico):
+        "Mirror of _asdict: take the dict and populate the tuple to be returned"
+        try:
+            obj = cls(**dico)
+        except TypeError as err:
+            logger.warning("TypeError: %s", err)
+            obj = cls(**{key: dico[key] for key in [i for i in cls._fields if i in dico]})
+        return obj
+
+    def as_dict(self):
+        """Like asdict, but possibly with some more features
+        """
+        return dict(asdict(self))
+
+    def __repr__(self):
+        return f"{self.name}: MotorRange({self.start:.3f}, {self.stop:.3f}, {self.points})"
+    @property
+    def step_size(self):
+        if self.points < 1:
+            return
+        return (self.stop-self.start)/self.points
+
+
+DataSetNT = namedtuple("DataSet", ("path", "h5", "nframes", "shape"), defaults=[None])
+
+@dataclass
+class DataSet:
+    path: str
+    h5: str = None
+    nframes: int = None
+    shape: tuple = None
+
+    def __repr__(self):
+        return f"Dataset('{self.path}', '{self.h5}', {self.nframes}, {self.shape})"
+
+    def as_tuple(self):
+        return DataSetNT(self.path, self.h5, self.nframes, self.shape)
+
+    @classmethod
+    def from_tuple(cls, tpl):
+        return cls(*tpl)
+
+    def is_hdf5(self):
+        """Return True if the object is hdf5"""
+        if self.h5 is None:
+            self.h5 = is_hdf5(self.path)
+        return bool(self.h5)
+
+    def __len__(self):
+        return self.nframes or 1
+
+
+class ListDataSet(list):
+
+    def commonroot(self):
+        """
+        :return: common directory
+        """
+        ll = [j.path.split(os.sep) for j in self]
+        common = os.path.commonprefix(ll)
```

Comment:
If you wish to return the common directory, you might want to use [commonpath](https://docs.python.org/3/library/os.path.html#os.path.commonpath) instead.